packages/components/src/components/linear-progress/linear-progress-container.jsx

```diff
@@ -2,61 +2,51 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { LinearProgress } from './linear-progress.jsx';
 
-class LinearProgressContainer extends React.Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            timeout: props.timeout / 1000,
-            total_ticks: Math.round(props.timeout / 1000),
-            current_tick: Math.round(props.timeout / 1000),
-        };
-    }
+const LinearProgressContainer = ({ timeout, action, render, ...other_props }) => {
+    const [timeout_state, setTimeout] = React.useState(timeout);
+    const [current_tick, setCurrentTick] = React.useState(Math.round(timeout / 1000));
+    const total_ticks = React.useState(Math.round(timeout / 1000));
 
-    get progress() {
-        return 100 - Math.round((this.state.current_tick / this.state.total_ticks) * 100);
-    }
+    const progress = () => {
+        return 100 - Math.round((current_tick / total_ticks) * 100);
+    };
```

Comment:
```suggestion
    const progress = () => 100 - Math.round((current_tick / total_ticks) * 100);
```

Reply:
Could we add a verb to the function name too? `getProgress`?

Reply:
I know it's not at the scope of this PR, but best time to do a refactor :pray:

Reply:
Changed as suggested